### ISSUE TITLE
Tpetra: Add TPETRA_USE_BLAS environment variable

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#684]
@trilinos/tpetra This depends on #654.  

The point of this issue is that Tpetra might not be able to trust the BLAS implementation to be threaded.  If TPETRA_USE_BLAS is true (has the value 1, or just plain is defined?), Tpetra should defer to the BLAS for GEMV and GEMM.  Otherwise, Tpetra should implement threading for these methods.  Tpetra may choose to call the underlying GEMV / GEMM implementation for single-thread kernels.

**Comment by kddevin** [comment]
Closed.  Won't fix.

**Comment by mhoemmen** [comment]
Clarifications:

  - Choice of what BLAS implementation to use belongs to kokkos-kernels.
  - Some of us hesitate to have too many obscure run-time controls for release-mode behavior.  

(Run-time options to help debug errors possibly involving Tpetra, are a different issue.)
